lib/puppet/rest/server_resolver.rb

```diff
@@ -0,0 +1,49 @@
+module Puppet::Rest
+  class ServerResolver
+
+    attr_reader :server, :port
+
+    # Return the appropriate server and port for the current settings
+    # @param [Symbol] srv_service the service to check for SRV records
+    # @param [String] default_server the fqdn of the server to fall back to
+    #                 if no other is found
+    # @param [Integer] default_port the port of the server to fall back to
+    #                  if no other is found
+    # @return [String, Integer] the server and port to use for the request
+    def select_server_and_port(srv_service: :puppet, default_server: nil, default_port: nil)
+      unless @server && @port
+        if Puppet.settings[:use_srv_records]
+          Puppet::Network::Resolver.each_srv_record(Puppet.settings[:srv_domain], srv_service) do |srv_server, srv_port|
```

Comment:
What relationship should this class have to `Puppet::Network::Resolver`? It looks like this and the original code in `Indirector::Request` are the only consumers of it? Should we update `Indirector::Request` to use this?

Reply:
Yeah updating `Indirector::Request` to use this should probably be pretty straightforward, and will allow to us to delete some duplicated tests as well.

I did a similar with `ResponseHandler`, but that one might be trickier because of the way we are exposing the `Net::HTTP` response object.